clang/lib/3C/DeclRewriter.cpp

```diff
@@ -26,6 +27,30 @@
 using namespace llvm;
 using namespace clang;
 
+RecordDecl *lastRecord = nullptr;
+std::map<Decl *, Decl *> VDToRDMap;
+std::set<Decl *> inlines;
```

Comment:
Are these global variables in `DeclRewriter`? Should they be `static`? We tend to want to avoid global variables.

Reply:
Maybe these belong in `ProgramInfo` ? @jackastner Suggestions?

Reply:
Or, maybe we don't need some of them at all, and they can be computed directly, as needed?

Reply:
I made these variables global in order to avoid having to pass them around between functions. If preferable, I can make them parameters in the functions that use these data structures. IIRC, there are only two such functions.

Reply:
I think it would be best to make this instance variables on the `DeclRewriter` class

Reply:
Sounds good.

Reply:
(And not making them `static`, if put in that class, makes sense to me.)

Reply:
I made these fields instance variables, but because I refer to them in `rewriteDecls` (which is a static function), I get errors asking me to use an object argument or make the helper function and associated fields static. Is there an obvious solution I'm missing?